packages/@lwc/integration-karma/test/wire/wiring/index.spec.js

```diff
@@ -21,7 +21,7 @@ function filterCalls(echoAdapterSpy, methodType) {
 
 describe('wiring', () => {
     describe('component lifecycle and wire adapter', () => {
-        it('should call a connect when component is connected', () => {
+        it('should call a connect when component is connected', async () => {
```

Comment:
The tests in this file use a shared global state. Most of the tests, but not all, have asynchronous behavior. In Karma, the state was properly cleaned for both sync and async tests. With web-test-runner, the state got polluted with the synchronous tests. Converting the sync tests to async functions (even with no inner `await`/promises) solved the problem, although I don't fully understand why.

Note that I added `async` to _all_ the functions, not just the sync ones, because that's best practice for anything that returns a promise.